### ISSUE TITLE
Unable to rename SecurityGroup in OpenStack

### ISSUE THREAD

**Issue by michalmedvecky** [hashicorp/terraform#12912]
TF is unable to rename security group, which is in use by another object (resource, vip, ...)

### Terraform Version

Terraform v0.9.0-dev (89acb5811a25e59e6a9d55cf5a2429846e3b5982+CHANGES)

### Affected Resource(s)
- openstack_networking_secgroup_v2

### Terraform Configuration Files

`sgrenametest.tf`:

```hcl
resource "openstack_networking_secgroup_v2" "test-sg" {
  name = "1234"
  description = "aaaa"
}

resource "openstack_networking_secgroup_rule_v2" "allow-something" {
  direction = "ingress"
  ethertype = "IPv4"
  security_group_id = "${openstack_networking_secgroup_v2.test-sg.id}"
  remote_group_id = "${openstack_networking_secgroup_v2.test-sg.id}"
}

resource "openstack_networking_port_v2" "test_vip" {
  depends_on = ["openstack_networking_secgroup_v2.test-sg"]
  name = "abcdef"
  network_id = "d3ebd619-3f2d-45cb-8e28-86da7cdf4cc5"
  admin_state_up = "true"
  security_group_ids = ["${openstack_networking_secgroup_v2.test-sg.id}"]
}

```

### Debug Output
https://gist.github.com/michalmedvecky/8cd9248b60102b0aac0df0137c3013e8

### Expected Behavior
Security group name should be changed

### Actual Behavior
Terraform fails

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. create sgrenametest.tf (see above)
2. `terraform apply`
3. modify `name` on line2 from `1234` to `12345`
4. `terraform apply`

### Important Factoids

Terraform correctly removes all security group rules tight to the SG, but does not remove it from the VIP. Then tries to delete SG what fails, because it's "in use".

**Comment by jtopjian** [comment]
@michalmedvecky Thanks for reporting this. It looks like updating the name of a security group is possible to do without recreation. Requiring the group to be recreated was most likely done because Gophercloud doesn't yet have support for group updates. 

I think the best long-term solution is to get that added. I'll work on this shortly.

**Comment by jtopjian** [comment]
Quick update: I have a patch into Gophercloud to add this support. Once it's merged, I'll get this into Terraform right away.

**Comment by stack72** [comment]
Closed via #14815
